h2quic/client.go

```diff
@@ -178,7 +178,7 @@ func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 		requestedGzip = true
 	}
 	// TODO: add support for trailers
-	endStream := !hasBody
+	endStream := !hasBody && req.Method != http.MethodConnect
```

Comment:
I don't see how a CONNECT with a `request.Body = nil` even makes sense. You won't be able to proxy any data after all.

Please have a look at the standard library H2 implementation here: https://github.com/golang/net/blob/master/http2/transport.go#L769. They don't need any special cases for CONNECT, and I'm not convinced we need one in QUIC.